File: javascript/trinary/trinary.js
var Trinary = function (trinary) {
  this.trinary = trinary;
};

Trinary.prototype.toDecimal = function () {
  var result = 0;
  for (var i = 0; i < this.trinary.length; i++) {
    if (this.trinary[i] == '0' || this.trinary[i] == '1' || this.trinary[i] == '2') {
      result += this.trinary[i] * Math.pow(3, this.trinary.length - (i + 1));
    } else {
      return 0;
    }
  }
  return result;
};

module.exports = Trinary;

File: csharp/trinary/Trinary.cs
﻿using System;

class Trinary
{
  public static int ToDecimal(string value)
  {
    var result = 0;
    var position = value.Length - 1;
    foreach (var c in value)
    {
      if (char.IsDigit(c))
      {
        var number = (int)char.GetNumericValue(c);
        if (number > 2)
          return 0;
        result += number * (int)Math.Pow(3, position);
        position--;
      }
      else
      {
        return 0;
      }
    }
    return result;
  }
}